config/runtime.exs

```diff
@@ -0,0 +1,170 @@
+end
+
+# Only do some runtime configuration in production since in dev and tests the
+# files are automatically recompiled on the fly and thus, config/{dev,test}.exs
+# are just fine
+if config_env() == :prod do
+  # used for mailing, checking origins, finding tls certs…
+  domain_name = Teiserver.ConfigHelpers.get_env("DOMAIN_NAME", "beyondallreason.info")
+
+  certificates = [
+    keyfile: "/etc/letsencrypt/live/#{domain_name}/privkey.pem",
+    certfile: "/etc/letsencrypt/live/#{domain_name}/cert.pem",
+    cacertfile: "/etc/letsencrypt/live/#{domain_name}/fullchain.pem"
+  ]
+
+  # this is used in lib/teiserver_web/controllers/account/setup_controller.ex
+  # as a special endpoint to create the root user. Setting it to empty or nil
+  # will disable the functionality completely.
+  # There is already a root user, so disable it
+  config :teiserver, Teiserver.Setup, key: nil
+
+  config :teiserver, Teiserver,
+    game_name: "Beyond All Reason",
+    game_name_short: "BAR",
+    main_website: "https://www.beyondallreason.info/",
+    privacy_email: "privacy@beyondallreason.info",
+    discord: "https://discord.gg/beyond-all-reason",
+    ports: [
+      tcp: Teiserver.ConfigHelpers.get_env("SPRING_TCP_PORT", 8200, :int),
+      tls: Teiserver.ConfigHelpers.get_env("SPRING_TLS_PORT", 8201, :int),
+      # this can likely be deprecated and removed. It's for an old version
+      # of tachyon running on another TLS socket
+      tachyon: Teiserver.ConfigHelpers.get_env("TACHYON_TLS_PORT", 8202, :int)
+    ],
+    certs: certificates,
+    website: [
+      url: "beyondallreason.info"
+    ],
+    server_flag: "GB-WLS",
+    enable_benchmark: false,
+    node_name: Teiserver.ConfigHelpers.get_env("NODE_NAME"),
+    extra_logging: false,
+    enable_managed_lobbies: true,
+    user_agreement:
+      "A verification code has been sent to your email address. Please read our terms of service at https://#{domain_name}/privacy_policy and the code of conduct at https://www.beyondallreason.info/code-of-conduct. Then enter your six digit code below if you agree to the terms."
+
+  config :teiserver, Teiserver.Repo,
+    username: Teiserver.ConfigHelpers.get_env("DB_USERNAME"),
+    password: Teiserver.ConfigHelpers.get_env("DB_PASSWORD"),
+    database: Teiserver.ConfigHelpers.get_env("DB_NAME"),
+    pool_size: 40,
+    timeout: 120_000,
+    queue_interval: 2000
+
+  check_origin =
+    if Teiserver.ConfigHelpers.get_env("ACQUIRE_TLS_CERTIFICATE", false, :bool) do
+      ["//#{domain_name}", "//*.#{domain_name}"]
+    else
+      false
+    end
+
+  config :teiserver, TeiserverWeb.Endpoint,
+    url: [host: domain_name],
+    check_origin: check_origin,
+    https:
+      certificates ++
+        [
+          versions: [:"tlsv1.2"],
+          # dhfile is not supported for tls 1.3
+          # https://www.erlang.org/doc/man/ssl.html#type-dh_file
+          dhfile: Teiserver.ConfigHelpers.get_env("HTTPS_DH_FILE_PATH", "/etc/ssl/dhparam.pem")
```

Comment:
why this isn's together with `certificates` ?

Reply:
The certificates are required in two places, under `TeiserverWeb.Endpoint` and also under `Teiserver`. I'm not yet 100% sure why, I suspect it's because the `TeiserverWeb` is the html server for the website, and `Teiserver` is the thing handling lobbies and autohosts, but they both require the similar tls setup.

Reply:
Yes, I understand that, I'm just not clear why the dhparams is only for one of them, in all configs I've seen they are set in both.

Reply:
ah, I found where and how these are used. Here's [an example](https://github.com/geekingfrog/teiserver/blob/3b72d871c2b56101646b33418d7af5cd189c7e3b/lib/teiserver/tcp/spring/spring_tcp_server.ex#L19-L23).

I checked the available options, and I could indeed pull dhfile alongside.
I'd rather do that in another PR if possible, that one is already quite big and changing how we setup existing tcp/ssl pools seems out of scope for this one which purely aim at making the configuration more env-var driven.

If that's alright with you, I'll do these change later.

Reply:
Sounds good